app/src/main/java/io/pumpkinz/pumpkinreader/NewsDetailFragment.java

```diff
@@ -111,6 +116,20 @@ public void onViewCreated(View view, Bundle savedInstanceState) {
         } else if (news != null && newsDetailAdapter.hasLoadingMore()) {
             subscription = comments.subscribe(new CommentsSubscriber(false));
         }
+
+        refreshLayout = (SwipeRefreshLayout) view.findViewById(R.id.news_detail_refresh_container);
+        refreshLayout.setColorSchemeResources(R.color.pumpkin_accent);
+        refreshLayout.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
+            @Override
+            public void onRefresh() {
+                newsDetail.removeItemDecoration(itemDecoration);
```

Comment:
Better to do `forceUnsubscribe` here, to handle user refresh the comment again before the previous subscription is completed.

...or...

Disable the `refreshLayout` here and enable it again on `onCompleted` in `CommentsSubscriber`.